modin/core/dataframe/algebra/binary.py

```diff
@@ -413,3 +413,45 @@ def caller(
                 )
 
         return caller
+
+    @classmethod
+    def apply(
+        cls, left, right, func, axis=0, func_args=None, func_kwargs=None, **kwargs
+    ):
+        r"""
+        Apply a binary function row-wise or column-wise.
+
+        Parameters
+        ----------
+        left : modin.pandas.DataFrame or modin.pandas.Series
+            Left operand.
+        right : modin.pandas.DataFrame or modin.pandas.Series
+            Right operand.
+        func : callable(pandas.DataFrame, pandas.DataFrame, \*args, axis, \*\*kwargs) -> pandas.DataFrame
```

Comment:
the signature is wrong here, as the implementation explicitly passes Query Compilers as arguments...

Reply:
nope, `func` here is a kernel that will be applied to deserialized partitions (pandas dataframes) so the signature is correct

follow the track of the `func`:

1. [`register(func)`](https://github.com/dchigarev/modin/blob/d65387e6dbb7b894f6c3eb2c4936cacf190ac9a2/modin/core/dataframe/algebra/binary.py#L445)
2. `register(func)` -> [`modin_dataframe.apply_full_axis(func, ...`)](https://github.com/dchigarev/modin/blob/d65387e6dbb7b894f6c3eb2c4936cacf190ac9a2/modin/core/dataframe/algebra/binary.py#L393-L395)